### ISSUE TITLE
ASANA

### ISSUE THREAD

**Issue by pipedream-component-development** [PipedreamHQ/pipedream#10477]
### App
ASANA

### Summary:
Unable to select a Project within the workspace

### Details:
I am trying to connect to ASANA/Workspace/Project. I am able to connect to ASANA and the Workspace, but when clicking in the dropdown for Project, I get an error of "Cannot read properties of undefined (reading 'data')" Pipedream will not show me even one project. This obviously prohibits me from continuing on.

As a general note, I am new to Automation and I am currently working through Zapier, Make & now Pipedream to explore functionality, ease of use, limitations, and support to see if one platform is better than the other. 

Thanks for your help in advance.

### Screenshots:

![Screenshot](https://storage.tally.so/private/Screenshot-2024-02-20-092841.png?id=OaWKBk&accessToken=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6Ik9hV0tCayIsImZvcm1JZCI6Im5yampSdiIsImlhdCI6MTcwODQ0Njg4NX0.jErAcie4zMPELBrmZDWStpEB1ls-Cga1xqsaf49lfLM&signature=828453f594911ed92ac9f992d8c6bef61493153cc8d581355c1562ce6a86ff88)

**Comment by jhildreth01** [comment]
I also tried to Create a Source and received the following error when trying to find the Project.

{"errors":[{"message":"The result is too large. You should use pagination (may require specifying a workspace)!","help":"For more information on API status codes and how to handle them, read the docs on errors: https://developers.asana.com/docs/errors"}]}

![Screenshot 2024-02-20 093642](https://github.com/PipedreamHQ/pipedream/assets/160640454/73710b41-18d1-40e4-ac9f-b3023d13366a)

**Comment by jhildreth01** [comment]
Hey there team. Thanks for looking into this. I tried this again tonight and ran into another error. The short story is, I still cannot see any options or search for my Project. Thoughts?
![Screenshot 2024-02-27 202657](https://github.com/PipedreamHQ/pipedream/assets/160640454/0dc5dd48-2fff-4a6b-91db-bafc099fc704)

**Comment by michelle0927** [comment]
Hi @jhildreth01. Would you mind trying the source labeled “New Task (Instant)” instead? 

@dannyroosevelt @malexanderlim The source labeled “Task Added To Project (Instant)” isn’t present in the repo. It should probably be unpublished or redirected to “New Task (Instant)”.

**Comment by jhildreth01** [comment]
> Hi @jhildreth01. Would you mind trying the source labeled “New Task (Instant)” instead?
> 
> @dannyroosevelt @malexanderlim The source labeled “Task Added To Project (Instant)” isn’t present in the repo. It should probably be unpublished or redirected to “New Task (Instant)”.

When I tried this, I wasn't able to get past the "Project" selection. It gave me the following: 
![image](https://github.com/PipedreamHQ/pipedream/assets/160640454/44210365-0a12-4ec5-ad52-bedaba06fbfe)

**Comment by michelle0927** [comment]
@jhildreth01 I created another PR [here](https://github.com/PipedreamHQ/pipedream/pull/10630) that I think will solve the issue.

**Comment by malexanderlim** [comment]
Confirmed the the legacy trigger has been unpublished and redirected.

**Comment by jhildreth01** [comment]
I was able to go in and try this again, but I still ended up with the same error when trying "New Task (Instant)". I also tried it using "Task Updated In Project (Instant)", "New Story Added To Project (Instant)" and ran into the same issue with all three triggers.

When I used another ASANA identity, I was able to select the Workspace and Project. The Caviaet to the second identity is that it is pulling from a whole different ASANA team, so I know it works. 

For my Original "identity", I do know that it has been used for years, so the sheer volume of projects and tasks is quite staggering. The particular project I am trying to access contains over 525K tasks.
